src/NumericalAlgorithms/DiscontinuousGalerkin/BoundarySchemes/StrongFirstOrder/BoundaryFlux.hpp

```diff
@@ -0,0 +1,92 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/BoundarySchemes/StrongFirstOrder/BoundaryData.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/LiftFlux.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/NumericalFluxes/NumericalFluxHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Protocols.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace dg {
+/// Functionality related to the strong first-order boundary scheme.
+namespace StrongFirstOrder {
+
+/*!
+ * \brief Compute the boundary flux contribution and lift it to the volume.
+ *
+ * This computes the numerical flux minus the flux dotted into the interface
+ * normal, projects the result to the face mesh if necessary, and then lifts it
+ * to the volume (still presented only on the face mesh as all other points are
+ * zero).
+ *
+ * Projection must happen after the numerical flux calculation so that the
+ * elements on either side of the mortar calculate the same result.  Projection
+ * must happen before flux lifting because we want the factor of the magnitude
+ * of the unit normal added during the lift to cancel the Jacobian factor in
+ * integrals to preserve conservation; this only happens if the two operations
+ * are done on the same grid.
+ */
+template <size_t FaceDim, typename NumericalFluxType,
+          Requires<tt::conforms_to_v<NumericalFluxType,
+                                     dg::protocols::NumericalFlux>> = nullptr>
+Variables<typename NumericalFluxType::variables_tags> boundary_flux(
```

Comment:
I'll probably be renaming functions like this to `boundary_correction_term` because for nonconservative systems it's not a flux.